src/pages/server-sitemap.xml/index.tsx

```diff
@@ -1,20 +1,15 @@
 import { getServerSideSitemap } from 'next-sitemap'
 import { GetServerSideProps, GetServerSidePropsContext } from 'next'
+import { allDocsUrls, allTutorialsUrls } from 'lib/sitemap'
 
 export const getServerSideProps: GetServerSideProps = async (
 	ctx: GetServerSidePropsContext
 ) => {
-	const response = await fetch(
-		`https://content.hashicorp.com/api/all-docs-paths`
-	)
-
-	const { result } = await response.json()
-	const fields = result.map((page: { path: string; created_at: string }) => ({
-		loc: `${__config.dev_dot.canonical_base_url}/${page.path}`,
-		lastmod: page.created_at,
-		priority: 0.7,
-		changefreq: 'daily',
-	}))
+	// returns an array of docs content sitemap objects per slug
+	const docsUrls = await allDocsUrls()
+	// returns an array of tutorials content sitemap objects per slug
+	const tutorialsUrls = await allTutorialsUrls()
+	const fields = [...docsUrls, ...tutorialsUrls]
```

Comment:
suggestion: We can parallelize these calls to speed up the generation:
```suggestion
	// returns an array of docs content sitemap objects per slug
	const docsUrlsPromise = allDocsUrls()
	// returns an array of tutorials content sitemap objects per slug
	const tutorialsUrlsPromise = allTutorialsUrls()
	
	const [docsUrls, tutorialsUrls] = await Promise.all([
		docsUrlsPromise,
		tutorialsUrlsPromise,
	])
	
	const fields = [...docsUrls, ...tutorialsUrls]
```

Reply:
Good idea!